File: backend/src/main/java/com/deliveryplatform/service/AgencyCustomerService.java
package com.deliveryplatform.service;

import com.deliveryplatform.dto.request.AgencyCustomerRequest;
import com.deliveryplatform.dto.response.AgencyCustomerResponse;
import com.deliveryplatform.domain.entity.AgencyCustomerStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Map;
import java.util.UUID;

public interface AgencyCustomerService {
    Page<AgencyCustomerResponse> getCustomers(UUID agencyId, String query, Pageable pageable);
    AgencyCustomerResponse getCustomerDetails(UUID agencyId, UUID customerId);
    AgencyCustomerResponse createCustomer(UUID agencyId, AgencyCustomerRequest request);
    AgencyCustomerResponse updateCustomer(UUID agencyId, UUID customerId, AgencyCustomerRequest request);
    void updateStatus(UUID agencyId, UUID customerId, AgencyCustomerStatus status);
    void deleteCustomer(UUID agencyId, UUID customerId);
    Map<String, Object> getAnalytics(UUID agencyId);
}

File: frontend/src/services/api/agencyCustomerService.ts
import apiClient from '@/api/client';
import { ENDPOINTS } from '@/api/endpoints';

const GET_ENDPOINTS = (agencyId: string) => ENDPOINTS.AGENCIES.CUSTOMERS;

export interface AgencyCustomer {
  id: string;
  fullName: string;
  companyName?: string;
  email: string;
  phone: string;
  city?: string;
  address?: string;
  notes?: string;
  status: 'ACTIVE' | 'SUSPENDED' | 'BLOCKED';
  totalOrders: number;
  totalRevenue: number;
  successRate: number;
  createdAt: string;
  updatedAt: string;
  isVip: boolean;
  isHighRisk: boolean;
}

export interface AgencyCustomerRequest {
  fullName: string;
  companyName?: string;
  email: string;
  phone: string;
  city?: string;
  address?: string;
  notes?: string;
}

export const agencyCustomerService = {
  getCustomers: async (agencyId: string, query?: string, page = 0, size = 10) => {
    const response = await apiClient.get(ENDPOINTS.AGENCIES.CUSTOMERS.BASE(agencyId), {
      params: { query, page, size }
    });
    return response.data;
  },

  getCustomer: async (agencyId: string, customerId: string) => {
    const response = await apiClient.get(ENDPOINTS.AGENCIES.CUSTOMERS.BY_ID(agencyId, customerId));
    return response.data;
  },

  createCustomer: async (agencyId: string, data: AgencyCustomerRequest) => {
    const response = await apiClient.post(ENDPOINTS.AGENCIES.CUSTOMERS.BASE(agencyId), data);
    return response.data;
  },

  updateCustomer: async (agencyId: string, customerId: string, data: AgencyCustomerRequest) => {
    const response = await apiClient.put(ENDPOINTS.AGENCIES.CUSTOMERS.BY_ID(agencyId, customerId), data);
    return response.data;
  },

  suspendCustomer: async (agencyId: string, customerId: string) => {
    await apiClient.patch(ENDPOINTS.AGENCIES.CUSTOMERS.ACTION(agencyId, customerId, 'suspend'));
  },

  blockCustomer: async (agencyId: string, customerId: string) => {
    await apiClient.patch(ENDPOINTS.AGENCIES.CUSTOMERS.ACTION(agencyId, customerId, 'block'));
  },

  activateCustomer: async (agencyId: string, customerId: string) => {
    await apiClient.patch(ENDPOINTS.AGENCIES.CUSTOMERS.ACTION(agencyId, customerId, 'activate'));
  },

  deleteCustomer: async (agencyId: string, customerId: string) => {
    await apiClient.delete(ENDPOINTS.AGENCIES.CUSTOMERS.BY_ID(agencyId, customerId));
  },

  getAnalytics: async (agencyId: string) => {
    const response = await apiClient.get(ENDPOINTS.AGENCIES.CUSTOMERS.ANALYTICS(agencyId));
    return response.data;
  }
};
